agent/consul/peering_backend.go

```diff
@@ -205,23 +205,38 @@ func meshGatewayAdresses(state *state.Store, ws memdb.WatchSet, wan bool) ([]str
 	return addrs, nil
 }
 
+func parseNodeAddr(node *structs.ServiceNode) string {
+	// Prefer the wan address
+	if v, ok := node.TaggedAddresses[structs.TaggedAddressWANIPv4]; ok {
```

Comment:
Just a note - I don't make use of `advertise_addr_wan_ipv6` (`ipv6.disable=1`), but others may; what should the preference/priority be between `advertise_addr_wan`, `advertise_addr_wan_ipv4`, and `advertise_addr_wan_ipv6`?

Reply:
That's a good question, will see about how that preference could be communicated in a follow-up PR.

For now will defer to the parsed `TaggedAddressWAN`, which could be IPv6.